### PR TITLE
Utilize atexit making sure connection info is deleted

### PR DESCRIPTION
**Pull request by oysteoh**

### CODE REVIEW

**Comment by oysteoh** [comment]
Jenkins: Test this please!

File: ert_shared/services/_base_service.py [hunk 1]
```diff
@@ -53,6 +54,15 @@ def local_exec_args(script_args: Union[str, List[str]]) -> List[str]:
     return [sys.executable, str(Path(__file__).parent / script), *rest]
 
 
+def delete_connection_info(service_name: str) -> None:
+    """
+    Ensure that the JSON connection information file is deleted
+    """
+    file = Path(f"{service_name}_server.json")
+    if file.exists():
+        file.unlink()
```
Comment (dotfloat) [inline ert_shared/services/_base_service.py:63]: Maybe something like to unregister?:
```suggestion
        file.unlink()
    atexit.unregister(delete_connection_info)
```

Comment (oysteoh) [reply ert_shared/services/_base_service.py:63]: nop, no difference

File: ert_shared/services/_base_service.py [hunk 2]
```diff
@@ -32,6 +34,17 @@
 ConnInfo = Union[Mapping[str, Any], Exception, None]
 
 
+SERVICE_NAMES: Set[str] = set()
+
+
+@atexit.register
+def cleanup_service_files():
+    for service_name in SERVICE_NAMES:
+        file = Path(f"{service_name}_server.json")
```
Comment (mortalisk) [inline ert_shared/services/_base_service.py:43]: this says "server", while the test says "service"

File: tests/ert_tests/services/test_base_service.py [hunk 3]
```diff
@@ -301,3 +303,26 @@ def test_local_exec_args_multi():
     assert exec_args[0] == sys.executable
     assert exec_args[2] == "foo"
     assert exec_args[3] == "-bar"
+
+
+def test_cleanup_service_files(tmpdir):
+    global SERVICE_NAMES
+
```
Comment (frode-aarstad) [inline tests/ert_tests/services/test_base_service.py:310]: Perhaps this global should be deleted or cleaned up so it does not mess up other tests?

**Review by frode-aarstad — APPROVED** [review]
LGTM

**Pull request merged**